### ISSUE TITLE
Date is not saved on Power off

### ISSUE THREAD

**Issue by EvilDragon** [tdm/android_device_fxtec_pro1#23]
If you power off your phone and don't have any internet connection when you boot up, at least the date is off (didn't check if the time was okay or not).

When I booted up the phone today in an airplane to listen to music, the date was set to January 31st.

**Comment by tdm** [comment]
I cannot reproduce this issue on my devices.  One running stock, one running Lineage.  On both devices: (1) Set airplane mode, (2) Power off, (3) Power on.  Both kept the correct date.

In any case, date and time does need an internet connection to function properly due to hardware lacking an RTC.  If the device has a SIM card, the date and time should be fetched from the cell network.  If no SIM card is present, the date and time should be fetched over the internet via public NTP servers.

**Comment by jannoke** [comment]
Tested on stock also.  True - time does not have timezone applied when entering in pin dialogue. It 2 hours off in here, but after loading into launcher clock is also correct in airplane mode. Have only one phone so have not installed lineage yet.

**Comment by EskeRahn** [comment]
I can reproduce this issue on my device. Running Lineage Test6.

**The Short:**
Aeroplanemode is enough

**The Long:**
(0) no SIM, (1) Turned WiFi Off, (2) Set aeroplane mode, (3) Power off, (4) waited five minutes (5) Power on. Both time and date where off, more precisely two days and roughly half an hour (time of install I think). 
As soon as I activated the WiFi, the stamp was corrected,

I then retried, and almost immediately turned it back on, and again same time 2020-02-18 22:18 CET

I then retried just selecting aeroplane mode and booting, and date/time was set back. Disabling Aeroplane mode, and had the time back within seconds.

Finally tried just a restart in Aeroplane mode, and that is enough to get the issue


**Suggestion:** 
A few seconds after boot, check if the stamp is close to the install-stamp, and if so present the user with the date-time dialogue. This really is a LOS issue, unless other devices usually have a RTC.

**Comment by tdm** [comment]
@EskeRahn no, I'm not aware of any qcom devices with an RTC.  But also Lineage doesn't really mess with time services -- these are generally provided by qcom and used as stock blobs.  So I'm not sure what I can do about it.  And it's probably not worth much effort in any case, as a device without connectivity is probably not a common use case.

**Comment by Loader009** [comment]
There might be a workaround, but this would require some modifications, probably not supported by this.
Using `touch /etc/banner` right before a reboot/shutdown would set the time right after a reboot.

This solution is actually used on OpenWRT for routers, which also doesn't have a RTC:
https://openwrt.org/docs/guide-user/base-system/cron#periodic_reboot
